packages/flutter/lib/src/material/page.dart

```diff
@@ -205,6 +202,8 @@ class MaterialPageRoute<T> extends PageRoute<T> {
     backGestureController?.dispose();
   }
 
+  bool get _useCupertinoTransition => false;
```

Comment:
I think we should only do the back gesture in cases where that's how we'd animate, unless we hear otherwise from the UX people. You can't use the back gesture for every page even in Apple apps.